beamer/chain.py

```diff
@@ -405,13 +405,20 @@ def maybe_withdraw(claim: Claim, context: Context) -> None:
     # This gives us a chronological order. The agent should never garbage collect
     # a request which has active claims
     assert request is not None, "Active claim for non-existent request"
+    assert context.address in {claim.claimer, claim.challenger}, "Agent is not part of claim"
 
     block = context.latest_blocks[request.source_chain_id]
     if block["timestamp"] < claim.termination:
-        # TODO: once L1 resolution is implemented, maybe withdraw before claim is terminated
-        return
+        # If the agent is the L1 filler, we can withdraw
+        # Condition is inverted, as the withdraw happens below
+        if request.l1_resolution_filler != context.address:
+            return
 
-    agent_winning = claim.get_winning_address() == context.address
+        # TODO: what if the non of the participants the the l1 filler
+
+    # The L1 resolution has precedence over the challenge game
+    claim_winner = request.l1_resolution_filler or claim.get_winning_address()
```

Comment:
the agent could also be the winner if he is not the `l1_resolution_filler`. 
If the agent is the challenger the winning condition for him is `claim.claimer != l1_resolution_filler`
If he is the claimer the winning condition is `claim.claimer == context.address`